### ISSUE TITLE
Service registered - cast vs. FSM event

### ISSUE THREAD

**Issue by rzezeski** [basho/riak_api#37]
I'm seeing the following in my logs while running Yokozuna tests.

```
2013-09-30 18:06:52.665 [error] <0.955.0>@riak_api_pb_server:handle_info:325 Unrecognized message {'$gen_cast',{registered,yz_pb_search}}
2013-09-30 18:06:52.666 [error] <0.955.0>@riak_api_pb_server:handle_info:325 Unrecognized message {'$gen_cast',{registered,yz_pb_admin}}
```

It seems the sup sends a cast but the PB server is an FSM and wants an event.

https://github.com/basho/riak_api/blob/develop/src/riak_api_pb_server.erl#L261

https://github.com/basho/riak_api/blob/develop/src/riak_api_pb_sup.erl#L43

I would think there should be a `registered/2` API on `riak_api_pb_server` to avoid situations like this.

**Comment by rzezeski** [comment]
Still seeing these errors in latest test runs.

```
2013-11-23 16:16:29.004 [error] <0.678.0>@riak_api_pb_server:handle_info:328 Unrecognized message {'$gen_cast',{registered,yz_pb_search}}
2013-11-23 16:16:29.014 [error] <0.678.0>@riak_api_pb_server:handle_info:328 Unrecognized message {'$gen_cast',{registered,yz_pb_admin}}
```

**Comment by seancribbs** [comment]
This is fixed by #44, just needs a reviewer.
